models/issue_comment.go

```diff
@@ -553,32 +553,81 @@ func (c *Comment) CodeCommentURL() string {
 
 // LoadPushCommits Load push commits
 func (c *Comment) LoadPushCommits() (err error) {
-	if c.Content == "" {
-		return err
+	if c.Content == "" || c.Commits != nil || c.Type != CommentTypePullPush {
+		return nil
 	}
 
 	commitIDs := strings.Split(c.Content, ":")
-	if int64(len(commitIDs)) != c.Line {
-		return fmt.Errorf("LoadPushCommits: len of commitIDs is wrong %d - %d",
-			len(commitIDs),
-			c.Line)
-	}
-	c.Commits, err = getCommitsFromCommitIDs(c.Issue.Repo, commitIDs)
-	if err != nil {
-		return err
-	}
-
-	c.Commits = ValidateCommitsWithEmails(c.Commits)
-	c.Commits = ParseCommitsWithSignature(c.Commits, c.Issue.Repo)
-	c.Commits = ParseCommitsWithStatus(c.Commits, c.Issue.Repo)
 
 	if c.IsForcePush {
 		c.OldCommit = commitIDs[0]
 		c.NewCommit = commitIDs[1]
+	} else {
+		repoPath := c.Issue.Repo.RepoPath()
+		gitRepo, err := git.OpenRepository(repoPath)
+		if err != nil {
+			return err
+		}
+		defer gitRepo.Close()
+
+		c.Commits = gitRepo.GetCommitsFromIDs(commitIDs)
+		c.Commits = ValidateCommitsWithEmails(c.Commits)
+		c.Commits = ParseCommitsWithSignature(c.Commits, c.Issue.Repo)
+		c.Commits = ParseCommitsWithStatus(c.Commits, c.Issue.Repo)
 	}
+
 	return err
 }
 
+// LoadPullPushDefaultNotify load default notify message in Content
+// with Mardown style for Pull Push commits comment
+func (c *Comment) LoadPullPushDefaultNotify() (err error) {
+	if c.Type != CommentTypePullPush {
+		return nil
+	}
+	if err = c.LoadIssue(); err != nil {
+		return
+	}
+	if err = c.Issue.LoadRepo(); err != nil {
+		return
+	}
+	if err = c.Issue.LoadPullRequest(); err != nil {
+		return
+	}
+
+	message := "@" + c.Poster.Name
+	if c.IsForcePush {
+		commitIDs := strings.Split(c.Content, ":")
+		message += " force-pushed the " + c.Issue.PullRequest.HeadBranch + " branch from " + commitIDs[0] + " to " + commitIDs[1]
+	} else {
+		if c.Commits == nil {
+			repoPath := c.Issue.Repo.RepoPath()
+			gitRepo, err := git.OpenRepository(repoPath)
+			if err != nil {
+				return err
+			}
+			defer gitRepo.Close()
+			commitIDs := strings.Split(c.Content, ":")
+			c.Commits = gitRepo.GetCommitsFromIDs(commitIDs)
+		}
+
+		if c.Commits.Len() == 1 {
+			message += fmt.Sprintf(" pushed 1 commit to %s:  \n ", c.Issue.Repo.Name)
```

Comment:
Alternative: prepare json content with the relevant data and save that in the comment content, then retrieve that in the template, so translations can be used. E.g.:
```
{
    "forced": true,
    "commits": [
        "abcdef": "Commit message line 1\nline 2\netc.",
        "ghijklm": "Other commit message line 1\nline 2\netc.",
    ]
}
```

Reply:
Column size should be checked, though (in any solution).

Reply:
> Column size should be checked, though (in any solution).

I'm sorry, but I can't fully understand your meaning, do you means limit output cmmits number? Thanks

Reply:
> Alternative: prepare json content with the relevant data and save that in the comment content, then retrieve that in the template, so translations can be used. E.g.:
> 
> ```
> {
>     "forced": true,
>     "commits": [
>         "abcdef": "Commit message line 1\nline 2\netc.",
>         "ghijklm": "Other commit message line 1\nline 2\netc.",
>     ]
> }
> ```

Hmm, If use this way, output defaut message (user haven't define custom template) will become hard ..., maybe should do some change on default template ..., I don't know whether It's a good idea, I will try it. Thanks

Reply:
I meant that the code in `routers/` should expand this before sending the info to the template. The router has the proper lang context, so it can use translations.

Reply:
> 
> 
> > Column size should be checked, though (in any solution).
> 
> I'm sorry, but I can't fully understand your meaning, do you means limit output cmmits number? Thanks

I meant I don't remember the max number of bytes/chars that we can fit into the `Comment.Content` column, and that's DB dependent.